modules/agent/pkg/controllers/bundledeployment/controller.go

```diff
@@ -111,12 +111,17 @@ func (h *handler) DeployBundle(bd *fleet.BundleDeployment, status fleet.BundleDe
 func (h *handler) checkDependency(bd *fleet.BundleDeployment) (string, bool, error) {
 	bundleNamespace := bd.Labels["fleet.cattle.io/bundle-namespace"]
 	for _, depend := range bd.Spec.DependsOn {
-		ls := &metav1.LabelSelector{
-			MatchLabels: map[string]string{
-				"fleet.cattle.io/bundle-name":      depend.Name,
-				"fleet.cattle.io/bundle-namespace": bundleNamespace,
-			},
+		ls := &metav1.LabelSelector{}
+
+		if depend.BundleSelector != nil {
+			ls = depend.BundleSelector
+		}
+
+		if depend.Name != "" {
+			ls = metav1.AddLabelToSelector(ls, "fleet.cattle.io/bundle-name", depend.Name)
+			ls = metav1.AddLabelToSelector(ls, "fleet.cattle.io/bundle-namespace", bundleNamespace)
 		}
```

Comment:
Will line 117's "depend.BundleSelector" get overriden by lines 121 and 122? It appears not to be the case, but just checking.

Reply:
yeah the labels for name and namespace will be appended to the the provided BundleSelector